### PR TITLE
add Junction boundary type for HDMap

### PR DESCRIPTION
**Pull request by msbeta**
1.add Junction boundary type for HDMap;
2.fix problem reported by security team;

### CODE REVIEW

File: modules/map/hdmap/adapter/xml_parser/junctions_xml_parser.cc [hunk 1]
```diff
@@ -40,6 +40,7 @@ Status JunctionsXmlParser::Parse(const tinyxml2::XMLElement& xml_node,
       return Status(apollo::common::ErrorCode::HDMAP_DATA_ERROR, err_msg);
     }
 
+
```
Comment (xiaoxq) [inline modules/map/hdmap/adapter/xml_parser/junctions_xml_parser.cc]: Don't need this blank line.

File: modules/map/hdmap/adapter/xml_parser/junctions_xml_parser.cc [hunk 2]
```diff
@@ -48,16 +49,36 @@ Status JunctionsXmlParser::Parse(const tinyxml2::XMLElement& xml_node,
       return Status(apollo::common::ErrorCode::HDMAP_DATA_ERROR, err_msg);
     }
 
+    std::string junction_type;
+    checker =
+      UtilXmlParser::QueryStringAttribute(
+        *junction_node, "type", &junction_type);
```
Comment (xiaoxq) [inline modules/map/hdmap/adapter/xml_parser/junctions_xml_parser.cc]: Please try `scripts/clang-format.sh` to correct obvious style and layout issues.

File: modules/map/hdmap/adapter/xml_parser/junctions_xml_parser.cc [hunk 3]
```diff
@@ -84,6 +104,28 @@ Status JunctionsXmlParser::Parse(const tinyxml2::XMLElement& xml_node,
   return Status::OK();
 }
 
+Status JunctionsXmlParser::ToPbJunctionType(
+        const std::string &junction_type, PbJunctionType *type) {
+  CHECK_NOTNULL(type);
+
+  *type = apollo::hdmap::Junction_Type_UNKNOWN;
+  if (junction_type == "UNKONW") {
```
Comment (xiaoxq) [inline modules/map/hdmap/adapter/xml_parser/junctions_xml_parser.cc]: Is it a typo?

File: modules/map/hdmap/adapter/xml_parser/junctions_xml_parser.h [hunk 4]
```diff
@@ -15,9 +15,9 @@ limitations under the License.
 #pragma once
 
 #include <vector>
+#include <string>
```
Comment (xiaoxq) [inline modules/map/hdmap/adapter/xml_parser/junctions_xml_parser.h]: Should be in order.

File: modules/map/hdmap/adapter/xml_parser/junctions_xml_parser.h [hunk 5]
```diff
@@ -29,6 +29,8 @@ class JunctionsXmlParser {
  public:
   static Status Parse(const tinyxml2::XMLElement& xml_node,
                       std::vector<JunctionInternal>* junctions);
+  static Status ToPbJunctionType(const std::string &juntion_type,
+                      PbJunctionType* type);
```
Comment (xiaoxq) [inline modules/map/hdmap/adapter/xml_parser/junctions_xml_parser.h]: Ditto. Please try the clang-format tool.

File: modules/map/hdmap/adapter/xml_parser/junctions_xml_parser.cc [hunk 6]
```diff
@@ -72,8 +93,7 @@ Status JunctionsXmlParser::Parse(const tinyxml2::XMLElement& xml_node,
         OverlapWithJunction overlap_with_juntion;
         overlap_with_juntion.object_id = object_id;
         junction_internal.overlap_with_junctions.push_back(
-            overlap_with_juntion);
-
+          overlap_with_juntion);
```
Comment (xiaoxq) [inline modules/map/hdmap/adapter/xml_parser/junctions_xml_parser.cc]: Ditto. Please try the clang-format tool.

File: modules/map/hdmap/adapter/xml_parser/objects_xml_parser.cc [hunk 7]
```diff
@@ -158,9 +158,10 @@ Status ObjectsXmlParser::ParseStopLines(
       CHECK(curve_segment != nullptr);
       const auto sub_node = object_node->FirstChildElement("geometry");
       if (sub_node == nullptr) {
-        std::string err_msg = "Error parse stopline geometry";
-        return Status(apollo::common::ErrorCode::HDMAP_DATA_ERROR, err_msg);
+       std::string err_msg = "Error parse stopline geometry";
+       return Status(apollo::common::ErrorCode::HDMAP_DATA_ERROR, err_msg);
```
Comment (xiaoxq) [inline modules/map/hdmap/adapter/xml_parser/objects_xml_parser.cc]: Ditto. Please try the clang-format tool.

File: modules/map/hdmap/adapter/xml_parser/objects_xml_parser.h [hunk 8]
```diff
@@ -49,7 +49,7 @@ class ObjectsXmlParser {
                                 const std::string& child_node_name,
                                 std::vector<std::string>* passage_node_ids);
   static Status ToPassageType(const std::string& type,
-                              PbPassageType* passage_type);
+                                PbPassageType* passage_type);
```
Comment (xiaoxq) [inline modules/map/hdmap/adapter/xml_parser/objects_xml_parser.h]: Ditto. Please try the clang-format tool.

File: modules/map/hdmap/adapter/xml_parser/objects_xml_parser.cc [hunk 9]
```diff
@@ -362,6 +363,7 @@ Status ObjectsXmlParser::ToPassageType(const std::string& type,
   return Status::OK();
 }
 
+
```
Comment (xiaoxq) [inline modules/map/hdmap/adapter/xml_parser/objects_xml_parser.cc]: Don't need this line.

File: modules/map/hdmap/adapter/xml_parser/util_xml_parser.cc [hunk 10]
```diff
@@ -209,6 +209,120 @@ tinyxml2::XMLError UtilXmlParser::QueryStringAttribute(
   *value = val;
   return tinyxml2::XML_SUCCESS;
 }
+Status UtilXmlParser::ParseBoundaryPolygon(const tinyxml2::XMLElement& xml_node,
+                                  PbBoundaryPolygon* boundary_polygon) {
```
Comment (xiaoxq) [inline modules/map/hdmap/adapter/xml_parser/util_xml_parser.cc]: Ditto. Please try the clang-format tool.
As well as all other places.

File: modules/map/proto/map_road.proto [hunk 11]
```diff
@@ -5,16 +5,36 @@ package apollo.hdmap;
...
+  WALL = 4;
+  SUNKEN = 5;
+  POST_FENCE= 6;
+  VIRTUAL = 7;
+};
+
+message BoundaryEdgeTypeInfo {
+ optional double start_s = 1;
+ optional double end_s = 2;
+ repeated BoundaryEdgeType type = 3;
```
Comment (xiaoxq) [inline modules/map/proto/map_road.proto]: Indents as 2.

File: modules/map/proto/map_junction.proto [hunk 12]
```diff
@@ -4,6 +4,11 @@ package apollo.hdmap;
 
 import "modules/map/proto/map_id.proto";
 import "modules/map/proto/map_geometry.proto";
+import "modules/map/proto/map_road.proto";
```
Comment (xiaoxq) [inline modules/map/proto/map_junction.proto]: Make the header section sorted.

File: modules/map/hdmap/adapter/xml_parser/junctions_xml_parser.cc (hunk 1 above)
Comment (msbeta) [reply modules/map/hdmap/adapter/xml_parser/junctions_xml_parser.cc]: I have corrected code styles with the clang format tool。

**Pull request closed without merging**